semanticdb/metacp/src/main/scala/scala/meta/internal/scalacp/Scalacp.scala

```diff
@@ -435,11 +434,34 @@ object Scalacp {
           false
       }
     }
+    def isScalacField: Boolean = {
+      val isField = sym.isInstanceOf[MethodSymbol] && !sym.isMethod && !sym.isParam
+      val isJavaDefined = sym.isJava
+      isField && !isJavaDefined
+    }
+    def isUselessField: Boolean = {
+      val peers = sym.parent.map(_.children.toList).getOrElse(Nil)
```

Comment:
⚠️ handwavy

This seems like a heavy hammer for a small task, and on a hot path